### PR TITLE
Make js compiler happy

### PR DESCRIPTION
**Pull request by ebidel**
R: @aliafshar, @crhym3 

This is for g3 import. Makes js compiler :smile:

### CODE REVIEW

**Comment by rictic** [comment]
(I'll leave some inline comments explaining the changes which might otherwise appear kinda weird)

File: analytics-behavior.html [hunk 1]
```diff
@@ -15,14 +15,14 @@
 -->
 
 <script>
+
+  var GoogleCodelabs = window.GoogleCodelabs || {};
```
Comment (rictic) [inline analytics-behavior.html]: This is so that the `@polymerBehavior` declaration is next to the implementation

File: analytics-behavior.html [hunk 2]
```diff
@@ -37,7 +37,7 @@
       params.hitType = hit;
       params.dimension1 = target.environment;
       params.dimension2 = target.category;
-      window.ga(function() {
+      window['ga'](function() {
```
Comment (rictic) [inline analytics-behavior.html]: The official google analytics compiler externs don't understand this method of calling window.ga, so I hid it from the compiler with a string literal access

File: google-codelab-survey.html [hunk 3]
```diff
@@ -86,7 +86,7 @@
       /**
        * Fired when user answers a survey question.
        * @event google-codelab-survey-answer
-       * @detail {{question: String, answer: String}}
+       * detail {{question: String, answer: String}}
```
Comment (rictic) [inline google-codelab-survey.html]: `@detail` is heresy! No, actually, it's just that the compiler doesn't like unknown tags

File: google-codelab-survey.html [hunk 4]
```diff
@@ -95,7 +95,10 @@
          * It is important to set this to a non-empty string
          * when hosting 2+ codelabs/surveys on the same domain.
          */
-        surveyId: '',
+        surveyId: {
+          type: String,
```
Comment (rictic) [inline google-codelab-survey.html]: Compiler likes types very much yes

File: google-codelab-survey.html [hunk 5]
```diff
@@ -131,7 +134,7 @@
       _storageLoaded: function() {
         var elems = this.getContentChildren();
         var answer = '';
-        for (var i = 0; elem = elems[i]; i++) {
+        for (var i = 0, elem; elem = elems[i]; i++) {
```
Comment (rictic) [inline google-codelab-survey.html]: Actual bug! Undeclared variable, would fail in strict mode

File: google-codelab-survey.html [hunk 6]
```diff
@@ -143,6 +146,10 @@
         }
       },
 
+      /**
+       * @param {?} groupEl .
+       * @param {?=} value .
```
Comment (rictic) [inline google-codelab-survey.html]: Needed to explain that `value` was an optional parameter

File: google-codelab.html [hunk 7]
```diff
@@ -177,7 +177,7 @@ <h1 class="title">{{title}}</h1>
       /**
        * Fired when user advances a codelab step or goes backwards.
        * @event google-codelab-step
-       * @detail {{step: Object, index: Number}}
+       * detail {{step: Object, index: Number}}
```
Comment (rictic) [inline google-codelab.html]: `@detail` heresy

File: google-codelab.html [hunk 8]
```diff
@@ -350,7 +350,7 @@ <h1 class="title">{{title}}</h1>
       ready: function() {
         // TODO: cannot be done on prototype.
         // SEe https://github.com/PolymerElements/iron-a11y-keys-behavior/issues/19
-        this.keyEventTarget = document.body;
+        this.keyEventTarget = /** @type {!HTMLBodyElement} */(document.body);
```
Comment (rictic) [inline google-codelab.html]: Did you know? `document.body` can be null!

I did not know this. Needed to explain to the compiler that it's never null by the time we're in this ready callback.

File: google-codelab.html [hunk 9]
```diff
@@ -410,7 +410,7 @@ <h1 class="title">{{title}}</h1>
         var step = this.steps[this.selected];
         step.active = true;
 
-        location.hash = this.selected;
+        location.hash = this.selected.toString();
```
Comment (rictic) [inline google-codelab.html]: numbers and strings living together, oh my!

File: google-codelab.html [hunk 10]
```diff
@@ -447,7 +447,7 @@ <h1 class="title">{{title}}</h1>
       },
 
       _updateStepIndexFromUrl: function() {
-        var step = parseInt(location.hash.slice(1));
+        var step = parseInt(location.hash.slice(1), 10);
```
Comment (rictic) [inline google-codelab.html]: parseInt behaves less unpredictably when you're explicit about the base. Probably for the best

File: google-codelab-survey.html (hunk 5 above)
Comment (ebidel) [inline google-codelab-survey.html]: Yea, this one was on us :)

**Comment by crhym3** [comment]
Awesome :star2:
I wonder if we should add a closure compiler pass as a CI task to at least make sure it stays compatible for future changes.

**Comment by ebidel** [comment]
Sure. Will file a bug for that though.

**Comment by ebidel** [comment]
https://github.com/googlecodelabs/codelab-components/issues/21

**Pull request merged**